cmd/singularity/cli/singularity.go

```diff
@@ -175,9 +175,6 @@ func sylabsToken(cmd *cobra.Command, args []string) {
 	if authToken == "" {
 		authToken, authWarning = auth.ReadToken(defaultTokenFile)
 	}
-	if authToken == "" && authWarning == auth.WarningTokenFileNotFound {
-		sylog.Warningf("%v : Only pulls of public images will succeed", authWarning)
-	}
```

Comment:
`authWarning` is set but never used, meaning `WarningTokenFileNotFound` (or any other `authWarning` value is not output to the user in any scenario.)

It would be better to move this function to `internal/pkg/util/auth`, and have it return the token and an `error`. Then call directly from the Library/Builder/Keystore commands that use it. That way, you could output the specific reason with the `FATAL` line.

Reply:
If this means that we can control the particular fatal error output from the server side,... I'd give it a huge +10.